src/chapters/angular-lsn3/exercises.rst

```diff
@@ -11,31 +12,255 @@ The starter code for the exercises is in the same
+      <img #rocketImage src="assets/images/LaunchCode_rocketline_white.png" height = "75" width = "75" [style.left]="0" [style.bottom]="0"/>
+
+While you're here, add the click handler to the *Right* button:
+
+.. sourcecode:: html+ng2
+      
+      <button (click)="handleRightClick(rocketImage)">Right</button>
+
+Now in ``app.component.ts`` we can write the ``handleRightClick()``:
+
+.. sourcecode:: TypeScript
+   :linenos:
+
+   handleRightClick(rocketImage) {
+      let movement = parseInt(rocketImage.style.left) + 10 + 'px';
+      rocketImage.style.left = movement;
+   }
+
+Follow the same pattern for the other directional buttons, modifying the
+movement formula as needed. For the *Up* and *Down* buttons, we'll want
+to also update the current status object's height property.
+
+For example, the *Down* click handler could look like this:
+
+.. sourcecode:: TypeScript
+   :linenos:
+
+   handleDownClick(rocketImage) {
+    let movement = parseInt(rocketImage.style.bottom) - 10 + 'px';
+    rocketImage.style.bottom = movement;
+    this.status.height = this.status.height - 10000;
+   }
+
+Update the Control Button Click Handlers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Along those same lines, we'll want to modify a couple of our control 
+button handlers to update ``rocketImage``'s position when the status
+changes. Pass in ``rocketImage`` to your *Land* and *Abort Mission* 
+handlers and add the following:
+
+.. sourcecode:: TypeScript
+
+   rocketImage.style.bottom = '0px';
+
+New Requirements
 ----------------
+
+You'll notice, a user can move the rocket before the shuttle is officially taken off. One can also
+abort a mission while the rocket is landed. This doesn't make much sense. With attribute
+directives, we can dynamically set those buttons to only be enabled in some states.
+
+Let's add some properties to our status objects to account for the enabled/disabled status of our
+control buttons. 
+
+.. sourcecode:: TypeScript
+   :linenos:
+
+   initStatus = {
+    color: 'green',
+    height: 0,
+    message: 'Space shuttle ready for takeoff!',
+    takeOffEnabled: true,
+    landEnabled: false,
+    missionAbortEnabled: false
+   };
+
+When the app is first loaded, we want the user to be able to initiate the ``shuttleTakeOff`` status,
+but not ``shuttleLand`` or ``shuttleMissionAbort``. We'll add some ``[disabled]`` attribute
+directives on the control buttons to reflect these values.
+
+Update the control buttons:
+
+.. sourcecode:: html+ng2
+   :linenos:
+      
+   <div class="container-control-buttons">
+      <button (click)="handleTakeOff()" [disabled]="!status.takeOffEnabled">Take Off</button>
+      <button (click)="handleLand(rocketImage)" [disabled]="!status.landEnabled">Land</button>
+      <button (click)="handleMissionAbort(rocketImage)" [disabled]="!status.missionAbortEnabled">Abort Mission</button>
+   </div>
+
+Now, only the *Take Off* button is enabled and the other two controls are disabled on first load,
```

Comment:
Comma fixes: "Now only the *Take Off* button is enabled, and the other..."

Alternatively: "Now, based on the boolean values we added to ``initStatus``, only the *Take Off* control button is enabled when the rocket is on the ground."